src/HalfLine.cpp

```diff
@@ -0,0 +1,17 @@
+//
+// Created by haico on 22-04-20.
+//
+
+#include "HalfLine.h"
+
+namespace rtt {
+    std::optional<Vector2> HalfLine::intersect(Line line) {
+        auto result = Line::generalIntersect(ending, direction, line.start, line.end);
+        if (result.has_value()) {
+            float t = std::get<1>(result.value());
+            if (t >= 0) return std::optional(std::get<0>(result.value()));
+        }
+        return std::nullopt;
+    }
```

Comment:
I still don't like this kind of function using the general intersect method, but I know you will disagree so let's not waste time discussing this.

Reply:
I do not agree with both of you, since generalIntersect in the Line class will now also be useable by the LineSegment class due to these changes. This will eliminate all forms of code clones.

Reply:
You are tying together functionality that is not necessary to tie together. I strongly believe this makes everything a lot less clear, even if you say it reduces code clones. It is fine if you do not agree with me, but I will not merge it until you split the functionality.

Reply:
Well then there will not be any HalfLine class